### PR TITLE
Improve Java/Jni IM read/subscribe

### PR DESCRIPTION
**Pull request by yunhanw-google**
-- Add auto-resubscribe capability for subscribe
-- Refactor read/subscribe with better exception check. 
-- manually validate the resubscription via bring up/down/up all-cluster-app
-- when receiving custom cluster, report would suppress the error generated from clusterObjects.
-- Fix platform::delete usage in jni
-- add imTimeout parameter for read/subscribe
-- disable cluster state data chace in java/jni and only use event
cache, version cache from cluster state cache, since java also have
cache to store the data

### CODE REVIEW

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt [hunk 1]
```diff
@@ -179,7 +179,8 @@ class WildcardFragment : Fragment() {
                                        minInterval,
                                        maxInterval,
                                        keepSubscriptions,
-                                       isFabricFiltered)
+                                       isFabricFiltered,
+                                       0)
```
Comment (g-coppock) [inline examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt:183]: Can you add inline comments for these zeroes? e.g.

`/* imTimeoutMs= */ 0`

here and below? That will make it clearer what that zero represents.

**Review by g-coppock — APPROVED** [review]

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java [hunk 2]
```diff
@@ -539,14 +539,27 @@ public byte[] getAttestationChallenge(long devicePtr) {
...
+  /**
+   * @brief Auto-Resubscribe to the given attribute path with keepSubscriptions and isFabricFiltered
+   * @param SubscriptionEstablishedCallback Callback when a subscribe response has been received and
+   *     processed
+   * @param ResubscriptionAttemptCallback Callback when a resubscirption haoppens, the termination
+   *     cause is provided to help inform subsequent re-subscription logic.
+   * @param ReportCallback Callback when a report data has been received and processed for the given
+   *     paths.
+   * @param devicePtr connected device pointer
+   * @param attributeList a list of attribute paths
```
Comment (joonhaengHeo) [inline src/controller/java/src/chip/devicecontroller/ChipDeviceController.java:551]: 1. Require to add minInterval, maxInterval.
2. Require to delete ResubscriptionAttemptCallback
3. attributeList -> attributePath

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java [hunk 3]
```diff
@@ -558,17 +571,31 @@ public void subscribeToAttributePath(
...
+   * @brief Auto-Resubscribe to the given event path with keepSubscriptions and isFabricFiltered
+   * @param SubscriptionEstablishedCallback Callback when a subscribe response has been received and
+   *     processed
+   * @param ResubscriptionAttemptCallback Callback when a resubscirption haoppens, the termination
+   *     cause is provided to help inform subsequent re-subscription logic.
+   * @param ReportCallback Callback when a report data has been received and processed for the given
+   *     paths.
+   * @param devicePtr connected device pointer
+   * @param eventList a list of event paths
+   * @param imTimeoutMs im interaction time out value, it would override the default value in c++ im
```
Comment (joonhaengHeo) [inline src/controller/java/src/chip/devicecontroller/ChipDeviceController.java:588]: 1. Require to add minInterval, maxInterval.
2. Require to delete ResubscriptionAttemptCallback
3. eventList -> eventPath

File: examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt (hunk 1 above)
Comment (yunhanw-google) [reply examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/WildcardFragment.kt]: done, thanks

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java (hunk 3 above)
Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/ChipDeviceController.java:592]: done, thanks

File: src/controller/java/src/chip/devicecontroller/ChipDeviceController.java (hunk 2 above)
Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/ChipDeviceController.java]: done, thanks

**Review by joonhaengHeo — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**